lib/pavilion/cmd_utils.py

```diff
@@ -380,7 +381,7 @@ def build_local(tests: List[TestRun],
 
         test_threads = [thr for thr in test_threads if thr is not None]
 
-        if cancel_event.is_set():
+        if cancel_event.is_set() and hard_fail is True:
```

Comment:
So... this won't work. 

Once the cancel event is set, all the builds with break regardless of if hard_fail is true. Builds watch for the cancel event and quit if they see it set. They also set it on failure.  

I actually came to annotate the ticket with info on how to deal with this. You need to:
1. Know when a build failed.
2. Have the primary thread cancel the builds when one fails, but only if hard_fail is True.

Currently both these steps are handled by the cancel event without intervention by the main thread. This will have to be separated. 
You can:
1. Have two events. A cancel_event to tell the builds to stop and a fail_event for the builds to tell the main thread they failed. 
2. Have one event still, but tell the main thread through other means such as by modifying the build info tracker.
    - You could also track this with the status file, but I don't recommend it.
3. Something more clever.